.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  mix_test:
+    name: mix test (Elixir ${{matrix.elixir}} | Erlang/OTP ${{matrix.otp}})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - elixir: 1.7.4
+            otp: 19.3.6.13
+          - elixir: 1.8.2
+            otp: 20.3.8.26
+          - elixir: 1.9.4
+            otp: 20.3.8.26
+            warnings_as_errors: true # not 1.10 as its --warnigs-as-errors has bugs https://github.com/elixir-lang/elixir/issues/10073
+          - elixir: 1.10.4
+            otp: 21.3.8.16
+          - elixir: 1.10.4
+            otp: 23.0.3
+            check_formatted: true
+    env:
+      MIX_ENV: test
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-elixir@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          elixir-version: ${{matrix.elixir}}
+      - name: Install Dependencies
+        run: |
+          mix local.hex --force
+          mix local.rebar --force
+          mix deps.get --only test
+      - run: mix format --check-formatted
+        if: matrix.check_formatted
+      - run: mix compile --warnings-as-errors
+        if: matrix.warnings_as_errors
+      - run: mix test
+
+  static_analysis:
+    name: static analysis (Elixir ${{matrix.elixir}} | Erlang/OTP ${{matrix.otp}})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - elixir: 1.10.4
+            otp: 23.0.3
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-elixir@v1
+        with:
+          otp-version: ${{matrix.otp}}
+          elixir-version: ${{matrix.elixir}}
+      - name: Cache build artifacts
+        uses: actions/cache@v2
+        with:
+          path: _build/dev/
+          key: ${{ matrix.otp }}-${{ matrix.elixir }}-static_analysis-build
+      - name: Install Dependencies
+        run: |
+          mix local.hex --force
+          mix local.rebar --force
+          mix deps.get
+      - name: Restore timestamps to prevent unnecessary recompilation
+        run: IFS=$'\n'; for f in $(git ls-files); do touch -d "$(git log -n 1 --pretty='%cI' -- $f)" "$f"; done
+      - run: mix credo --all --strict
+      - run: mix dialyzer
```

Comment:
You should cache the PLTs for speed

Reply:
They were already being cached since https://github.com/elixir-lsp/elixir_sense/pull/106/commits/be6871b2923e88ac0c6827bf1971b24dc0915e77. Bear in mind that "caches that are not accessed within the last week will be evicted" ([source](https://github.com/actions/cache#cache-limits)).

In any case, I extended the cache path list.

The following provide proof that the cache is working (an [empty commit](https://github.com/elixir-lsp/elixir_sense/pull/106/commits/4089be1867995350792ae9fbfdbb8f0b6202341a) apart):
- https://github.com/adrianomitre/elixir_sense/runs/1051989899 ← 9m 13s
- https://github.com/adrianomitre/elixir_sense/runs/1052052811 ← 12s